Open Judge System/Web/OJS.Web/Controllers/TempController.cs

```diff
@@ -368,5 +369,109 @@ public ActionResult MigrateSolutionSkeletons()
                 newLine +
                 $"{string.Join(newLine, exceptions.Select(ex => ex.ToString()).Distinct())}");
         }
+
+        public async Task<ActionResult> MigrateContestCategoryFromRemoteJudge(string id)
+        {
+            try
+            {
+                var contestCategoryResponse = await this.FetchContestCategory(int.Parse(id));
+
+                await this.LoadContestCategoryAndAssignCheckerAndSubmissionTypes(contestCategoryResponse.Data);
+
+                using (var scope = TransactionsHelper.CreateTransactionScope())
+                {
+                    this.Data.ContestCategories.Add(contestCategoryResponse.Data);
+                    this.Data.SaveChanges();
+                    scope.Complete();
+                    return this.Content($"The contest category is added!");
+                }
+            }
+            catch (Exception e)
+            {
+                return this.Content(
+                    $"Contest categories can't be migrated and exception {e}");
+            }
+        }
+
+        private async Task LoadContestCategoryAndAssignCheckerAndSubmissionTypes(ContestCategory contestCategory)
+        {
+            if (contestCategory == null)
+            {
+                return;
+            }
+
+            foreach (var contest in contestCategory.Contests)
+            {
+                foreach (var problemGroup in contest.ProblemGroups)
+                {
+                    this.LoadContestCategoryAndAssignCheckerAndSubmissionTypes(problemGroup.Problems);
+                }
+            }
+
+            var ids = contestCategory.Children.Select(x => x.Id).ToList();
+            contestCategory.Children.Clear();
+
+            foreach (var id in ids)
+            {
+                var contestCategoryResponse = await this.FetchContestCategory(id);
+
+                contestCategory.Children.Add(contestCategoryResponse.Data);
+                await this.LoadContestCategoryAndAssignCheckerAndSubmissionTypes(contestCategoryResponse.Data);
+            }
+        }
+
+        private async Task<ExternalDataRetrievalResult<ContestCategory>> FetchContestCategory(int id)
+        {
+            var contestCategoryResponse = await this.httpRequester.GetAsync<ContestCategory>(
+                new { id },
+                string.Format(UrlConstants.GetContestCategoryExportDataApiFormat, Settings.JudgeBaseUrl),
+                Settings.ApiKey);
+
+            if (!contestCategoryResponse.IsSuccess || contestCategoryResponse.Data == null)
+            {
+                throw new InvalidOperationException(contestCategoryResponse.ErrorMessage);
+            }
+
+            return contestCategoryResponse;
+        }
+
+        private void LoadContestCategoryAndAssignCheckerAndSubmissionTypes(IEnumerable<Problem> problems)
```

Comment:
I don't see anything related to loading contest categories in the method implementation. Why `LoadContestCategory` in the name?